tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -38,6 +38,63 @@
     ASSERT_TRUE([backingElement isKindOfClass:[WXFrameworkElement class]]);
 }
 
+@interface UIDeallocTestButton : UIButton {
+    std::shared_ptr<UXEvent> _event;
+}
+@end
+
+@implementation UIDeallocTestButton
+
+- (void)setDeallocEvent:(std::shared_ptr<UXEvent>)event {
+    _event = event;
+}
+
+-(void)dealloc {
+    _event->Set();
+    [super dealloc];
+}
+@end
+
+TEST(UIButton, CheckForLeaks) {
+    Microsoft::WRL::WeakRef weakXamlElement;
+    {
+        __block UIButtonWithControlsViewController* buttonVC = [[UIButtonWithControlsViewController alloc] init];
```

Comment:
>__block [](start = 8, length = 7)

I might have missed this before but do you really need __block here. I am under the impression, it should be retained correctly.